src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
@markstory I think that we need to create DecimalType instead of mapping to FloatType. Strictly, decimal values cannot be mapped to float values. So `id` is declared as `number` instead of `float` in the table.
Reading #9424, the `id` is declared as number(**20**, 0), but float types cannot keep such large precision. This code would work when id is 123456789012345, but wouldn't work when id is 12345678901234567.